map/bookmark_manager.cpp

```diff
@@ -1141,6 +1142,53 @@ kml::CompilationType BookmarkManager::GetCompilationType(kml::MarkGroupId id) co
   return compilation->second->GetCategoryData().m_type;
 }
 
+void BookmarkManager::SaveTrackRecording(std::string const & name)
+{
+  CHECK_THREAD_CHECKER(m_threadChecker, ());
```

Comment:
All the bookmark add/edit/delete operations are performed on the main thread and should be called from the main because the `bm manager` and `bm edit session` work only on the main.

The `SaveTrackRecording` method does not involve large file operations, only filling the track instance with data. The writing to the file will be handled as usual - at the end of the edit session and performed on the file thread - see `void BookmarkManager::NotifyChanges(bool saveChangesOnDisk)`.